enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,475 @@
+currently maintained as a part of the Ingress Operator.
+
+
+Here is an example of what an ExternalDNS deployment (created by the operator) would look like on AWS:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: external-dns
+  namespace: openshift-external-dns
+spec:
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: external-dns
+  template:
+    metadata:
+      labels:
+        app: external-dns
+    spec:
+      serviceAccountName: external-dns
+      containers:
+      - name: external-dns-private
+        image: k8s.gcr.io/external-dns/external-dns:v0.8.0
+        args:
+        - --source=openshift-route
+        - --provider=aws
+        # can only have one zone at a time if domains are overlapping over zones
+        - --aws-zone-type=private
+        - --registry=txt
+        - --txt-owner-id=externalDNS-default
+        - --log-level=debug
+        # Needs to be unique per container
+        - --metrics-address=:7979
+        - --zone-id-filter=/hostedzone/myzone
+        env:
+          - name: AWS_ACCESS_KEY_ID
+            value: abc
+          - name: AWS_SECRET_ACCESS_KEY
+            value: xyz
+      - name: external-dns-public
+        image: k8s.gcr.io/external-dns/external-dns:v0.8.0
+        args:
+        - --source=openshift-route
+        - --provider=aws
+        - --policy=upsert-only # would prevent ExternalDNS from deleting any records, omit to enable full synchronization
+        # can only have one zone at a time if domains are overlapping over zones
+        - --aws-zone-type=public
+        - --registry=txt
+        - --txt-owner-id=externalDNS-default
+        - --log-level=debug
+        # Needs to be unique per container
+        - --metrics-address=:7980
+        # Filter from aws dev account
+        - --zone-id-filter=/hostedzone/myzone
+        env:
+          - name: AWS_ACCESS_KEY_ID
+            value: abc
+          - name: AWS_SECRET_ACCESS_KEY
+            value: xyz
+```
+
+Note the use of multiple ExternalDNS containers within the same pod in the above deployment. Metrics listening ports would have
+to be configured on a per-container basis to avoid collisions.
+
+Metrics will be exposed via an insecure channel by default.
+On OpenShift, externalDNS instances can be deployed with a reverse proxy sidecar, such as kube-rbac-proxy, to provide a secure means of metrics scraping.
+
+### Risks and Mitigations
+
+Cluster administrators may not want to give users the ability to create arbitrary DNS records in an external provider when they grant users permission to create routes.
+ExternalDNS could observe a sub-field of a route resource when determining whether or not to publish records to the upstream provider.
+Cluster administrators could restrict a users ability to set this route sub-field, using a mechanism similar to that for route custom hosts.
+
+Similarly, cluster administrators may want to restrict the ability for users to specify the ExternalDNS
```

Comment:
Check with @deads2k on this.  I'd like to know what his opinion is.

Reply:
cc @Miciah